### PR TITLE
Taproot receive

### PR DESCRIPTION
**Pull request by benma**

### CODE REVIEW

**Comment by benma** [comment]
@thisconnect you will need a BitBox02 v9.10.0 to see taproot accounts. In this PR, taproot is only added to *new* accounts, so either delete accounts.json before, or add a new account to see it.

The receive screen now shows all the available address type options. Please fix up the styling :smile: The new locize keys have no text yet, we can discuss what to put there.

**Review by thisconnect — CHANGES_REQUESTED** [review]
made a comment to fix eth

File: frontends/web/src/routes/account/receive/receive.tsx [hunk 1]
```diff
@@ -60,12 +60,19 @@ interface LoadedReceiveProps {
 
 type Props = LoadedReceiveProps & ReceiveProps & TranslateProps;
 
+// For BTC/LTC: all possible address types we want to offer to the user, ordered by priority (first one is default).
+// Types that are not available in the addresses delivered by the backend should be ignored.
+const scriptTypes: accountApi.ScriptType[] = ['p2wpkh', 'p2tr', 'p2wpkh-p2sh']
+
 class Receive extends Component<Props, State> {
+    // All script types that are present in the addresses delivered by the backend. Will be empty for if there are no such addresses, e.g. in Ethereum.
+    private availableScriptTypes: accountApi.ScriptType[] = scriptTypes.filter(sc => this.props.receiveAddresses.findIndex(addrs => addrs.scriptType !== null ? sc === addrs.scriptType : -1) >= 0);
```
[the remark this answers is not in the document]
Comment (benma) [reply frontends/web/src/routes/account/receive/receive.tsx]: good catch! done

**Review by thisconnect — APPROVED** [review]
Got taproot addresses with v9.10.0 and deleted accounts.json 👍 

also tested current address types with current firmware and btc ltc eth 👍 

one weird case was when I tested same wallet with a different device that has the old firmware the Bitcoin accounts that received taproot didn't show. Adding did not let me add the first account.

other than that LGTM 👍

**Comment by benma** [comment]
> one weird case was when I tested same wallet with a different device that has the old firmware the Bitcoin accounts that received taproot didn't show. Adding did not let me add the first account.

Great testing, thanks! The account is missing due to this check:

https://github.com/digitalbitbox/bitbox-wallet-app/blob/2df942ab1955dbac02b0e6bf9b9bae0fdbbc955d/backend/accounts.go#L643-L645

Not sure what to do about it, if anything.

**Comment by benma** [comment]
@thisconnect your PR and my fixup didn't consider the order of the addresses in

```
const scriptTypes: accountApi.ScriptType[] = ['p2wpkh', 'p2tr', 'p2wpkh-p2sh']
```

I fixed it now to cycle through the types in that order.

**Pull request merged**